staging/src/k8s.io/client-go/tools/cache/mutation_cache.go

```diff
@@ -60,21 +61,23 @@ type ResourceVersionComparator interface {
 // If includeAdds is true, objects in the mutation cache will be returned even if they don't exist
 // in the underlying store. This is only safe if your use of the cache can handle mutation entries
 // remaining in the cache for up to ttl when mutations and deletes occur very closely in time.
-func NewIntegerResourceVersionMutationCache(backingCache Store, indexer Indexer, ttl time.Duration, includeAdds bool) MutationCache {
+func NewIntegerResourceVersionMutationCache(ctx context.Context, backingCache Store, indexer Indexer, ttl time.Duration, includeAdds bool) MutationCache {
```

Comment:
```
Incompatible changes:
 - ./tools/cache.NewIntegerResourceVersionMutationCache: changed from func(Store, Indexer, time.Duration, bool) MutationCache to func(context.Context, Store, Indexer, time.Duration, bool) MutationCache 
 ```
 
 https://grep.app/search?q=NewIntegerResourceVersionMutationCache
 
 Not used outside of Kubernetes?

Reply:
doesn't look like.

Reply:
This seems like an overkill, based on the discussion we've had with Jordan during KubeCon I believe we've agreed to apply context where it made sense. This cache seems like one that should not have, at most I'd expect it to only accept logger. This would resolve the other comment I had about creating a controller without the necessity for context.

Reply:
Ack. This is one of those places that I hadn't updated based on the "it's okay to put logr.Logger into our APIs" conclusion.

What could potentially throw a monkey wrench into this "no context during construction" approach is when other components don't follow it. We've seen that with event recorder. Yesterday I found that construction a [delaying queue](https://github.com/kubernetes/kubernetes/blob/8a39b6062060ffb4fba52f53c011bd00428db194/staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go#L138-L150) also already spawns a goroutine during `New*`.

I still think "no goroutines during construction" is a sound principle, sometimes it just might be a bit harder to follow.

Reply:
I took your `ctx` -> `logger` replacement for `NewIntegerResourceVersionMutationCache` from https://github.com/kubernetes/kubernetes/pull/129148 into this PR.

I did not take the other changes for controller construction. That can be a follow-up PR?

Reply:
> I still think "no goroutines during construction" is a sound principle, sometimes it just might be a bit harder to follow.

I don't want to clean those, yet. Probably in a separate PR, but in the long run we should definitely ensure that.